src/lib/services/namespaces-service.ts

```diff
@@ -8,11 +8,35 @@ import type { Settings } from '$lib/types/global';
 import { paginated } from '$lib/utilities/paginated';
 import { requestFromAPI } from '$lib/utilities/request-from-api';
 import { routeForApi } from '$lib/utilities/route-for-api';
+import {
+  toNamespaceArchivalStateReadable,
+  toNamespaceStateReadable,
+} from '$lib/utilities/screaming-enums';
 
 const emptyNamespace = {
   namespaces: [],
 };
 
+const toNamespaceDetails = (
+  namespace: DescribeNamespaceResponse,
+): DescribeNamespaceResponse => {
+  if (namespace.config) {
+    namespace.config.historyArchivalState = toNamespaceArchivalStateReadable(
+      namespace.config?.historyArchivalState,
+    );
+    namespace.config.visibilityArchivalState = toNamespaceArchivalStateReadable(
+      namespace.config?.visibilityArchivalState,
+    );
+  }
+
+  if (namespace.namespaceInfo) {
+    namespace.namespaceInfo.state = toNamespaceStateReadable(
+      namespace.namespaceInfo?.state,
```

Comment:
Can we remove the `?` here since there is the `namespace.namespaceInfo` check above (same with `namespace.config`)? Or maybe remove the check if `toNamespaceStateReadable` can handle `undefined`?

```suggestion
      namespace.namespaceInfo.state,
```

Reply:
👍 Yup good call